wrapper/src/main/java/software/amazon/jdbc/plugin/iam/IamAuthConnectionPlugin.java

```diff
@@ -45,7 +46,7 @@
 import software.amazon.jdbc.util.telemetry.TelemetryFactory;
 import software.amazon.jdbc.util.telemetry.TelemetryGauge;
 
-public class IamAuthConnectionPlugin extends AbstractConnectionPlugin {
+public class IamAuthConnectionPlugin extends AbstractConnectionPlugin implements AuthenticationConnectionPlugin {
```

Comment:
I think the FederatedAuthPlugin should also implement AuthenticationConnectionPlugin